tests/test_utils/test_ruleutils.py

```diff
@@ -1,41 +1,39 @@
-import os
-import unittest
+from pathlib import Path
+
+import pytest
 
 from linkml_runtime.dumpers import yaml_dumper
-from linkml_runtime.loaders.yaml_loader import YAMLLoader
 from linkml_runtime.utils.ruleutils import get_range_as_disjunction, subclass_to_rules
 from linkml_runtime.utils.schemaview import SchemaView
 from tests.test_utils import INPUT_DIR
 
-SCHEMA = os.path.join(INPUT_DIR, "rules-example.yaml")
-
-yaml_loader = YAMLLoader()
-
-
-class RuleUtilsTestCase(unittest.TestCase):
-    def test_disjunction(self):
-        # no import schema
-        view = SchemaView(SCHEMA)
-        analyte = view.induced_slot("analyte", "Sample")
-        # print(analyte)
-        # print(analyte.any_of)
-        disj = get_range_as_disjunction(analyte)
-        # print(disj)
-        self.assertCountEqual(disj, {"MissingValueEnum", "AnalyteEnum"})
-        for s in view.all_slots().values():
-            disj = get_range_as_disjunction(s)
-            print(f"{s.name} DISJ: {disj}")
-
-    def test_roll_up(self):
-        # no import schema
-        view = SchemaView(SCHEMA)
-        c = view.get_class("ProteinCodingGene")
-        rules = subclass_to_rules(view, "ProteinCodingGene", "SeqFeature")
-        rule = rules[0]
-        print(f"IF: {rule.preconditions}")
-        print(f"THEN: {rule.postconditions}")
-        print(yaml_dumper.dumps(rule))
-
-
-if __name__ == "__main__":
-    unittest.main()
+
+@pytest.fixture
+def rules_schema():
+    """SchemaView loaded with rules example schema."""
+    return SchemaView(Path(INPUT_DIR) / "rules-example.yaml")
+
+
+def test_disjunction(rules_schema):
+    """Test get_range_as_disjunction with schema view fixture."""
+    analyte = rules_schema.induced_slot("analyte", "Sample")
+    # print(analyte)
+    # print(analyte.any_of)
+    disj = get_range_as_disjunction(analyte)
+    # print(disj)
+    assert sorted(disj) == sorted({"MissingValueEnum", "AnalyteEnum"})
+
+    # Test all slots for debugging
+    for s in rules_schema.all_slots().values():
+        disj = get_range_as_disjunction(s)
+        print(f"{s.name} DISJ: {disj}")
+
+
+def test_roll_up(rules_schema):
+    """Test subclass_to_rules with schema view fixture."""
+    c = rules_schema.get_class("ProteinCodingGene")
+    rules = subclass_to_rules(rules_schema, "ProteinCodingGene", "SeqFeature")
+    rule = rules[0]
+    print(f"IF: {rule.preconditions}")
+    print(f"THEN: {rule.postconditions}")
+    print(yaml_dumper.dumps(rule))
```

Comment:
This isn't your fault, but this is a crap test; it isn't testing anything, it's just printing stuff. Can you work out what results one might expect here and put in an assertion?